### PR TITLE
[deep link] Set deepLinkValidation feature flag to true

### PR DESCRIPTION
**Pull request by hangyujin**
Launch has been approved.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or there is a reason for not adding tests.

![build.yaml badge]

If you need help, consider asking for help on [Discord].

[Contributor Guide]: https://github.com/flutter/devtools/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[build.yaml badge]: https://github.com/flutter/devtools/actions/workflows/build.yaml/badge.svg

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/feature_flags.dart [hunk 1]
```diff
@@ -62,7 +62,7 @@ abstract class FeatureFlags {
   /// DevTools screen and the standalone tool for IDE embedding.
   ///
   /// https://github.com/flutter/devtools/issues/6013
-  static bool deepLinkValidation = false;
```
Comment (chunhtai) [inline packages/devtools_app/lib/src/shared/feature_flags.dart:65]: This should be enableExperiments?

Comment (chunhtai) [reply packages/devtools_app/lib/src/shared/feature_flags.dart:65]: or probably `enableBeta`

Comment (chunhtai) [reply packages/devtools_app/lib/src/shared/feature_flags.dart:65]: talked to zoey offline, it is ok to release to all users now.

**Review by chunhtai — APPROVED** [review]

File: packages/devtools_app/release_notes/NEXT_RELEASE_NOTES.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ To learn more about DevTools, check out the
 
 ## General updates
 
-TODO: Remove this section if there are not any general updates.
+* Added a new feature for deep link validation, supporting deep link web checks on Android. - [#6935](https://github.com/flutter/devtools/pull/6935)
```
Comment (kenzieschmoll) [inline packages/devtools_app/release_notes/NEXT_RELEASE_NOTES.md:13]: Can we add a screenshot of this tool here?

Comment (kenzieschmoll) [reply packages/devtools_app/release_notes/NEXT_RELEASE_NOTES.md:13]: Discussed offline with Chun Heng. Will leave up to you whether you want to soft launch this or announce more loudly. These notes are displayed in DevTools when a user opens a new Devtools version for the first time.

**Review by kenzieschmoll — APPROVED** [review]

File: packages/devtools_app/release_notes/NEXT_RELEASE_NOTES.md (hunk 2 above)
Comment (chunhtai) [reply packages/devtools_app/release_notes/NEXT_RELEASE_NOTES.md:13]: I think this is fine to be a simple one line description just for the record. We can prepare more graphics and description once we are ready to do a public announcement.

**Pull request merged**